src/duct/__main__.py

```diff
@@ -472,13 +504,24 @@ def safe_close_files(file_list: Iterable[Any]) -> None:
             pass
 
 
-def ensure_directories(path: str) -> None:
+def ensure_directories(path: str, clobber: bool = False) -> None:
+    # Enforcing no path* files is perhaps overzealous, but this helps prevent
+    # conflicts between versions, prevents probable user errors, and leaves
+    # room for plugins down the road.
```

Comment:
@yarikoptic feedback on possible overzealousness requested :)

Reply:
I would have not used glob here but "explicit better than implicit" operated only on the file names we care to create/write.  For that I would centralize generation of those filenames, smth like some `@dataclass` structure with `.stdout, .stderr, .info, .stats` attributes with those filenames filled-in/expanded based on the prefix. Then you could go through all of them and verify that none of those particular files exist. That would also centralize logic of minting those filenames which is now "append some suffix and the point of use".

Reply:
Refactored out all the file handling, which touched nearly everything (and therefore proved that the old way was evil)